docs/wiki/development/building.md

```diff
@@ -171,6 +171,14 @@ cmake -G "Visual Studio 16 2019" -A x64 ..
 cmake --build . --config RelWithDebInfo -j10 # Number of projects to build in parallel
 ```
 
+### Optional: Configure project for Visual Studio IDE
+Configure with `-DADD_HEADERS_AS_SOURCES=ON`.
+Automatic CMake configuration may need to be disabled in Visual Studio via the **Tools > Options > CMake > General** menu.
+Then run the `prepare_for_ide` target to generate the necessary header symlinks and source files for the project.
```

Comment:
It's a CMake target that we have added so that if you haven't yet built anything, all the files that are normally generated at build time and that are needed for an IDE to work "correctly" (especially code completion), then are generated.
Mainly this creates all the symlinks to the public header files that the project needs.

Reply:
I'd suggest a slight wording tweak to reflect that this is a cmake command then.

Reply:
Yeah also both the CMake option `ADD_HEADERS_AS_SOURCES` and `prepare_for_ide` are available for all platforms.

The first one is optional since some IDEs are able to associate the headers with the rest/it won't impact completion, finding of references and so on, but in other cases it will.

In some other IDE's, it might create duplicate header items (which is why it's not on by default; it also doesn't exactly cover ALL headers that the project actually uses, since that's a bit more tricky and mainly a manual work to add them to the definition of each target).

Reply:
Happy to reword and/or relocate! Do you think it makes more sense to have a separate top-level IDE setup section? Is there any info on which IDEs require the `ADD_HEADERS_AS_SOURCES` option? If not I can at least test it out on some IDE on each platform. Almost seems like a separate wiki doc at that point though 😬